src/EzPlatformTemplatingBundle/lib/Core/EventListener/ContentViewSubscriber.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformTemplating\Core\EventListener;
+
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\MVC\Symfony\Event\PreContentViewEvent;
+use eZ\Publish\Core\MVC\Symfony\MVCEvents;
+use eZ\Publish\Core\MVC\Symfony\View\View;
+use EzSystems\EzPlatformTemplating\SPI\View\VariableProviderRegistry;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+
+final class ContentViewSubscriber implements EventSubscriberInterface
+{
+    private const EXPRESSION_INDICATOR = '@=';
+
+    public const PARAMETERS_KEY = 'params';
+
+    /** @var \EzSystems\EzPlatformTemplating\SPI\View\VariableProviderRegistry */
+    private $parameterProviderRegistry;
+
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    private $configResolver;
+
+    /** @var \Symfony\Component\ExpressionLanguage\ExpressionLanguage */
+    private $expressionLanguage;
+
+    public function __construct(
+        VariableProviderRegistry $parameterProviderRegistry,
+        ConfigResolverInterface $configResolver
+    ) {
+        $this->parameterProviderRegistry = $parameterProviderRegistry;
+        $this->configResolver = $configResolver;
+
+        $this->expressionLanguage = new ExpressionLanguage();
+        $this->expressionLanguage->register(
+            'provider',
+            function (string $identifier) {
+                return 'Not implemented: Not a Dependency Injection expression';
+            },
+            function (array $variables, string $identifier) {
+                /** @var \EzSystems\EzPlatformTemplating\SPI\View\VariableProvider $provider */
+                $provider = $variables['providerRegistry']->getTwigVariableProvider($identifier);
+
+                return $provider->getTwigVariables($variables['view'], $variables);
+            }
+        );
```

Comment:
IMHO should be moved outside of the `ContentViewSubscriber`.  There is at least one good reason for this: the same functionality (provider) should be available in `ezplatform-query-fieldtype` (and in general it could be resued there). 

A) Use Symfony recommended approach (ref.  https://symfony.com/doc/current/components/expression_language/extending.html)
```php
<?php

use Psr\Cache\CacheItemPoolInterface;
use Symfony\Component\ExpressionLanguage\ExpressionLanguage as BaseExpressionLanguage;

final class ExpressionLanguage extends BaseExpressionLanguage
{
    public function __construct(CacheItemPoolInterface $cache = null, array $providers = [])
    {
        array_unshift($providers, new ExpressionLanguageProvider());

        parent::__construct($cache, $providers);
    }
}
```

```php 
use Symfony\Component\ExpressionLanguage\ExpressionFunction;
use Symfony\Component\ExpressionLanguage\ExpressionFunctionProviderInterface;

final class ExpressionLanguageProvider implements ExpressionFunctionProviderInterface
{
    public function getFunctions(): array
    {
        return [
            new ExpressionFunction(
                'provider',
                function (string $identifier): string {
                    return 'Not implemented: Not a Dependency Injection expression';
                },
                function (array $variables, string $identifier) {
                    /** @var \EzSystems\EzPlatformTemplating\SPI\View\VariableProvider $provider */
                    $provider = $variables['providerRegistry']->getTwigVariableProvider($identifier);

                    return $provider->getTwigVariables($variables['view'], $variables);
                }
            ),
        ];
    }
}
```

B) Create a factory

```php 
<?php

use Symfony\Component\ExpressionLanguage\ExpressionLanguage;

final class ExpressionLanguageFactory
{
    public function createExpressionLanguage(): ExpressionLanguage
    {
        $expressionLanguage = new ExpressionLanguage();
        $expressionLanguage->register(
            'provider',
            function (string $identifier) {
                return 'Not implemented: Not a Dependency Injection expression';
            },
            function (array $variables, string $identifier) {
                /** @var \EzSystems\EzPlatformTemplating\SPI\View\VariableProvider $provider */
                $provider = $variables['providerRegistry']->getTwigVariableProvider($identifier);

                return $provider->getTwigVariables($variables['view'], $variables);
            }
        );

        return $expressionLanguage;
    }
}
```

In both scenarios `ExpressionLanuage` needs to be injected into `ContentViewSubscriber`.